gulpfile.js

```diff
@@ -65,3 +68,42 @@ gulp.task('pretest', function(done) {
 });
 
 gulp.task('default',['prepublish']);
+
+gulp.task('types', ['tsc'], function(done) {
+  var folder = 'built';
+  var files = ['ptor', 'browser', 'locators'];
+  var outputFile = path.resolve(folder, 'index.d.ts');
+  var contents = '';
+  files.forEach(file => {
+    contents += readFile(folder, file);
+  });
+
+  // add module declaration
+  contents += 'declare module "protractor" {\n';
+  contents += '  export = protractor\n';
+  contents += '}\n';
+
+  // remove files with d.ts
+  glob.sync(folder + '/**/*.d.ts').forEach(file => {
+    fs.unlinkSync(path.resolve(file));
+  });
+
+  // write contents to 'built/index.d.ts'
+  fs.writeFileSync(outputFile, contents);
```

Comment:
Honestly, I think it's a little cleaner to just do:

`fs.writeFileSync('built/index.d.ts', contents)` and forget about keeping the file/output folder nicely in variables.